src/virtualenv/create/via_global_ref/builtin/python2/site.py

```diff
@@ -161,4 +162,23 @@ def add_global_site_package():
         site.PREFIXES = orig_prefixes
 
 
+# Debian and it's derivatives patch this function. We undo the damage
+def rewrite_getsitepackages():
+    site = sys.modules["site"]
+
+    orig_getsitepackages = site.getsitepackages
+
+    def getsitepackages():
+        sitepackages = orig_getsitepackages()
+        for prefix in (sys.prefix, sys.exec_prefix):
+            # os is not defined here, but will be in site.py, where we monkey-patch this function.
+            path = os.path.join(prefix, "lib", "python" + sys.version[:3], "site-packages")  # noqa: F821
```

Comment:
You can't write code like this, that's platform dependent. You must detect if you're on debian and execute this only then.

Reply:
Oh. Right. I forgot the check. I think checking for linux should be enough though, as non debian linux will already contain the path, so `if path not in sitepackages` will be False.

Reply:
The code now does the same check the debians patched site.py does. This prevents the code from running on systems where it would break.

It will still run on non-debian systems, but shouldn't have any effect there. I don't think specifically detecting debian would be practical.

Reply:
> Oh. Right. I forgot the check. I think checking for linux should be enough though, as non debian linux will already contain the path, so `if path not in sitepackages` will be False.

That's some very presumptions statement there. What stops macOS from patching similarly as does Debian? Or any other distribution? We really must restrict this to Debian only.

Reply:
I think that this should also handle all cases where an OS patches python in a way similar to debian.
The added path isn't debian specific. It is virtualenv specific and should always be in getsitepackages. The problem is just that because of debians patches it isn't.

The only thing I would have to fix is use `___EXPECTED_SITE_PACKAGES___` instead of hardcoding "site-packages".

I can however try restricting it to debian to be save.
Detecting debian and it derivatives isn't easy though. Just checking if `platform.dist` is debian isn't enough as we also need to detect Ubuntu and other derivatives.

After looking through all the debian patches I think the best way would be to detect the "deb_system" key in `distutils.command.install.INSTALL_SCHEMES`.
Alternatively it would be possible to search for occurrences of "dist-packages" in the source of getsitepackages.

Reply:
> The only thing I would have to fix is use `___EXPECTED_SITE_PACKAGES___` instead of hardcoding "site-packages".

This might suffice.

Reply:
If you can do this then 👍

Reply:
I just pushed a new version. I pulled some code from `load_host_site` into a new function to reuse it in `rewrite_getsitepackages`.

Because the code now makes no assumptions about the operating system other than what is already assumed by `load_host_site` I removed the os check.